### PR TITLE
cup2021

### PR DESCRIPTION
**Pull request by gdoffe**
Various fixes and configurations, mainly on LIDAR and sd21 in preparation of 2021 french robotic cup

### CODE REVIEW

**Review by ecourtois — CHANGES_REQUESTED** [review]
- In commit message "[drivers: sd21: remove size](https://github.com/cogip/mcu-firmware/pull/44/commits/ab92244197227e5ea741d2e704e32a22c4a559af)":
"that are written"

- In commit message "[platforms: cortex: fix SPI SD card reader configuration](https://github.com/cogip/mcu-firmware/pull/44/commits/5b31148d2537f6fc17830b54ab68ddea9bf5efaa)":
Two empty lines before Signed-off

CI does not pass the coding rules check.

Nice PR by the way!

File: platforms/cortex/platform.c [hunk 1]
```diff
@@ -263,9 +232,67 @@ static void *_task_planner_start_cancel(void *arg)
...
+
+    gpio_clear(GPIO_DEBUG_LED);
+
+    ctrl_set_anti_blocking_on(pf_get_ctrl(), TRUE);
+
+    /* Get platform path */
+    path_t *path = pf_get_path();
+    if (!path) {
+        printf("machine has no path\n");
+    }
```
Comment (Kheprys) [inline platforms/cortex/platform.c:279]: Should add an assert() to check path object here. The path is loaded right after on line 283 and can lead to a pointer exception.

Maybe previous gpio_init() checks can be done with assert() too but I don't know if it's a good idea or not

**Review by Kheprys — APPROVED** [review]
LGTM, maybe adding several assert() during init could be a good idea to avoid bad behavior

**Comment by gdoffe** [comment]
> * In commit message "[platforms: cortex: fix SPI SD card reader configuration](https://github.com/cogip/mcu-firmware/pull/44/commits/5b31148d2537f6fc17830b54ab68ddea9bf5efaa)":
>   Two empty lines before Signed-off

Strange :
image

File: platforms/pf_test/platform.c [hunk 2]
```diff
@@ -227,8 +227,8 @@ void pf_init(void)
 #endif /* MODULE_SHELL_PLATFORMS */
 
     /* Debug LED */
-    if (gpio_init(GPIO_DEBUG_LED, GPIO_OUT) == 0) {
-        puts("WARNING: GPIO_DEBUG_LED not initialized!");
+    if (gpio_init(GPIO_DEBUG_LED, GPIO_OUT)) {
+        puts("WARNING: GPIO_DEBUG_LED not initialized !");
```
[the remark this answers is not in the document]
Comment (ecourtois) [reply platforms/pf_test/platform.c:231]: Not fixed ;-)

**Review by ecourtois — CHANGES_REQUESTED** [review]
Just one extra space to remove and it's good!

**Review by ecourtois — APPROVED** [review]
LGTM!

**Pull request merged**